corehq/blobs/retry_s3db.py

```diff
@@ -0,0 +1,23 @@
+from botocore.exceptions import ClientError
+
+from dimagi.utils.retry import retry_on
+
+
+def retry_on_slow_down(func):
+    retry = retry_on(ClientError, should_retry=_is_slow_down,
+                     delays=_up_to_two_mins())
```

Comment:
This will create an iterable that can be iterated once, and then will always raise `StopIteration` (and yield no items) on subsequent uses, right? What do you think of just specifying the delays like so?
```suggestion
                     delays=[2, 4, 8, 16, 32, 64, 128, None])
```

Reply:
Hmmm, of course. Also, clearly, missing a test for that.

I kinda like being able to accept generators. What about ...
```python
        def retry(*args, **kw):
            more_delays = delays() if callable(delays) else delays
            for delay in chain(more_delays, [None]):
                # ...
```
... or would generators fall under YAGNI cos, like you suggest, we could just use a list here?

Reply:
I'm in the YAGNI camp